messages/ruby/lib/cucumber/messages/message/deserialization.rb

```diff
@@ -0,0 +1,142 @@
+require 'cucumber/messages/message/utils'
+require 'json'
+
+module Cucumber
+  module Messages
+    class Message
+      include Cucumber::Messages::Message::Utils
+
+      module Deserialization
+        def self.included(other)
+          other.extend(ClassMethods)
+        end
+
+        module ClassMethods
+
+          ##
+          # Returns a new Message - or messages into an array - deserialized from the given json document.
+          # CamelCased keys are properly converted to snake_cased attributes in the process
+          #
+          #   Cucumber::Messages::Duration.from_json('{"seconds":1,"nanos":42}')               # => #<Cucumber::Messages::Duration:0x00007efda134c290 @seconds=1, @nanos=42>
+          #   Cucumber::Messages::PickleTag.from_json('{"name":"foo","astNodeId":"abc-def"}')  # => #<Cucumber::Messages::PickleTag:0x00007efda138cdb8 @name="foo", @ast_node_id="abc-def">
+          #
+          # It is available on the base Cucumber::Messages::Message class as long as the
+          # keys in the json document match one of the DTO class available with the messages
+          #
+          #   Cucumber::Messages::Message.from_json('{"pickleTag":{"name":"foo","astNodeId":"abc-def"}}')  # => #<Cucumber::Messages::PickleTag:0x00007efda1451a00 @name="foo", @ast_node_id="abc-def">
+          #
+          # If more than one message is part of the JSON document, it will return an array of messages
+          #
+          #   Cucumber::Messages::Message.from_json('{
+          #     "pickleTag":{"name":"foo","astNodeId":"abc-def"},
+          #     "duration":{"seconds":1,"nanos":42}
+          #   }')
+          #   => [
```

Comment:
This surprising behaviour.

I would expect:

* JSON primitive => Ruby primitive
* JSON object => Ruby object
* JSON array => Ruby array

Here however, you're turning a JSON object into a Ruby array.